lib/resque/plugins/retry.rb

```diff
@@ -453,6 +453,13 @@ def on_failure_retry(exception, *args)
           return
         end
 
+        # If we are "ignoring" the exception, then we decrement the retry
+        # counter, so that the current attempt didn't count toward the retry
+        # counter.
+        if !@ignore_exceptions.nil? && @ignore_exceptions.include?(exception.class)
```

Comment:
You're right. I hadn't thought about that. Thank you for keeping me honest.

My concern was decrementing the counter and then still potentially
triggering the `try_again` or `give_up` callbacks. Can/should this be an
`if`, `elsif`, `else`?

Also, we should probably stick the result of the `decr` call into
`@retry_attempt`, no?
On Oct 26, 2015 6:13 AM, "Brandon Liu" <notifications@github.com> wrote:

> In lib/resque/plugins/retry.rb
> <https://github.com/lantins/resque-retry/pull/128#discussion_r42975885>:
>
> > @@ -453,6 +453,13 @@ def on_failure_retry(exception, *args)
> >            return
> >          end
> >
> > +        # If we are "ignoring" the exception, then we decrement the retry
> > +        # counter, so that the current attempt didn't count toward the retry
> > +        # counter.
> > +        if !@ignore_exceptions.nil? && @ignore_exceptions.include?(exception.class)
>
> Well at that moment we don't know yet what exception would have
> occurred... the equivalent solution is to move the increment logic "down
> here" so that the retry counter is incremented only after performing,
> rather than before.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lantins/resque-retry/pull/128/files#r42975885>.
>